src/index.ts

```diff
@@ -81,19 +82,21 @@ const RouterTitle = ({
 }: RouterTitleProps) => {
   const location = useLocation();
 
-  Promise.resolve(
-    callback(
-      getTitle(
-        routesConfig,
-        location.pathname,
-        divider,
-        pageTitle && [pageTitle],
+  useEffect(() => {
+    Promise.resolve(
+      callback(
+        getTitle(
+          routesConfig,
+          location.pathname,
+          divider,
+          pageTitle && [pageTitle],
+        ),
+        location,
       ),
-      location,
-    ),
-  ).then((title) => {
-    document.title = prefix ? `${prefix}${title}` : title;
-  });
+    ).then((title) => {
+      document.title = prefix ? `${prefix}${title}` : title;
+    });
+  }, [pageTitle, prefix, divider]);
```

Comment:
Are you sure about those rules. The `pageTitle` and `divider` should never become updated but the `location` will change.

Reply:
I removed all dependencies so it is updated on every render as it was before.